src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -112,7 +112,9 @@ bool SHAMap::walkBranch (SHAMapTreeNode* node,
     return true;
 }
 
-bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
+bool
+SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
+                 Delta& differences, int maxCount)
```

Comment:
`compare` genuinely calls things directly or indirectly that are non-const.  I did a full audit, making everything in `SHAMap` `const`, then only removing `const` when the compiler forced me to.